### PR TITLE
ENH: better LEEMDetector status checking

### PR DESCRIPTION
**Pull request by mrakitin**
This is a tentative enhancement for the existing `LEEMDetector` detector definition (thanks to @danielballan for the code).

Better way to do it would be after https://github.com/NSLS-II/ophyd/pull/630 is merged.

### CODE REVIEW

**Comment by mrakitin** [comment]
This is an old version of the code, and ESM-2 may have a better solution now. @ke-zhang-rd will be visiting the beamline next week, and can check with John if it's already implemented.

If not, @danielballan, do you mind to review it briefly next week?

File: startup/46-ESM_LEEM.py [hunk 1]
```diff
@@ -5,21 +5,30 @@
...
-        st = self.start_acq.set(1)
-        while not st.done:
-            # print(self.start_acq.value)
-            time.sleep(0.1)
-        while self.start_acq.value != 0:
-            # print(self.start_acq.value)
-            time.sleep(0.1)
-        # print(self.start_acq.value)
-        return NullStatus()
+        # On a background thread, listen for the server's response.
```
[the remark this answers is not in the document]
Comment (johnsinsheimer) [reply startup/46-ESM_LEEM.py]: I wrote this IOC such that start_acq is like an on/off button, with 0 off and 1 on, where on means acquisition in progress.  The transition from 1 to 0 is done by the IOC when acquisition is complete, on this same PV.

So I think "done" in this case is: " value == 0 and old_value == 1 " , opposite of the proposed change above.

**Comment by johnsinsheimer** [comment]
We're having some issues with this in that it doesn't actually wait for the acquisition to complete - it returns the prompt in IPython almost immediately, even though the "check_if_done" routine is still processing.  If we put a print statement in the "check_if_done" function, the run engine returns to the prompt and then later we see the print statement activate.

**Comment by mrakitin** [comment]
I pointed @gwbischof at a [similar solution](https://github.com/NSLS-II-QAS/profile_collection/blob/751e5164188dc9ec503ab813c5bc280d7384697d/startup/20-motors.py#L77-L103) at QAS earlier today. Have you tried it?

**Comment by johnsinsheimer** [comment]
I haven't had time to fully understand how everything is supposed to work here, but I wrote this below and tested today, and it worked the few times I tested it.  

Should I be doing .set or .put for the pv?  Is there a difference in this case?

Without this "init" variable or something equivalent, the second time the scan is run, it immediately finishes.

Also is there a better way to handle the subscription part than below?


```
#    def __init__(self, *args, **kwargs):
#        super().__init__(*args, **kwargs)
        # On a background thread, listen for the server's response.
#        self.st = None

    def trigger(self):
        init = 0
        status = DeviceStatus(self)
        # Write to server.
        def check_if_done(old_value, value, **kwargs):
            if init == 1:
                if value == 0 and old_value == 1:
                    status._finished()
                    self.acquire.clear_sub(check_if_done)

        self.acquire.subscribe(check_if_done)
        self.acquire.set(1)
        init = 1
        return status

```

**Comment by mrakitin** [comment]
@johnsinsheimer, I think we may still need the `__init__(...)` method to have access to the `self.st` status object, which you can use instead of a local var `init` in the `trigger(...)` method. Regarding the set/put question, the `set` is an asynchronous method, while the `put` is a synchronous one. In this case it's better to use `.put()` instead, or even better, `set_and_wait`:
```py
from ophyd.utils import set_and_wait
set_and_wait(self.acquire, 1)
```

It's a merged PR already. To facilitate a better discussion of the changes, do you mind to create a new pull request with the changes in a separate branch, so we can discuss the implementation there? Thanks.

**Pull request merged**